source/framework/CMakeLists.txt

```diff
@@ -21,6 +21,19 @@ if (NOT ${CURL_LIB} STREQUAL "CURL_LIB-NOTFOUND")
     set(external_libs "${external_libs};-lcurl;-lstdc++fs")
 endif ()
 
+find_path(MPFR_INCLUDE mpfr.h HINTS ${MPFR_PATH}) # path to include directory
+find_library(MPFR_LIB mpfr HINTS ${MPFR_PATH}) # path to .so file
+
+if (NOT MPFR_INCLUDE MATCHES "MPFR_INCLUDE-NOTFOUND" AND NOT MPFR_LIB MATCHES "MPFR_LIB-NOTFOUND")
+    # MPFR is found
+    add_compile_definitions(USE_MPFR)
+    set(external_include_dirs ${external_include_dirs} ${MPFR_INCLUDE})
+    set(external_libs "${external_libs};-lmpfr")
+    link_directories(${MPFR_LIB})
+else ()
+    message(WARNING "MPFR library was not found. Adding the location to the installation path may solve this problem.\n Use -DMPFR_PATH=/path/to/mpfr/")
```

Comment:
We should add some flags if this library is not linked, then one can type `rest-config --flags` to see if he can compile a MPFR-dependent REST library(like axionlib) based on this installation of REST